server/src/main/java/org/candlepin/pki/SubjectKeyIdentifierWriter.java

```diff
@@ -36,6 +40,14 @@
      * @throws IOException thrown if error reading cert
      * @throws NoSuchAlgorithmException thrown if JcaX509ExtensionUtils can't be created
      */
-    byte[] getSubjectKeyIdentifier(KeyPair clientKeyPair, Set<X509ExtensionWrapper> extensions)
-        throws IOException, NoSuchAlgorithmException;
+    default byte[] getSubjectKeyIdentifier(KeyPair clientKeyPair, Set<X509ExtensionWrapper> extensions)
+            throws IOException, NoSuchAlgorithmException {
+        return new JcaX509ExtensionUtils().createSubjectKeyIdentifier(clientKeyPair.getPublic()).getEncoded();
+    }
+
+    default byte[] toOctetString(byte[] bytes_to_encode) throws IOException {
+        DEROctetString octetStr = new DEROctetString(bytes_to_encode);
+        return octetStr.getEncoded(ASN1Encoding.DER);
```

Comment:
I don't think you need to specify ASN1Encoding.DER since that's implicit in the object type, DEROctetString.  Try without and see if the unit tests still pass.